src/units/abilities.cpp

```diff
@@ -698,9 +698,9 @@ std::pair<int,map_location> unit_ability_list::get_extremum(const std::string& k
 	int stack = 0;
 	for (const unit_ability& p : cfgs_)
 	{
-		int value = get_single_ability_value((*p.ability_cfg)[key], def, p, loc(), const_attack_ptr(), [&](const wfl::formula& formula, wfl::map_formula_callable& callable) {
-			return formula.evaluate(callable).as_int();
-		});
+		int value = std::round(get_single_ability_value((*p.ability_cfg)[key], static_cast<double>(def), p, loc(), const_attack_ptr(), [&](const wfl::formula& formula, wfl::map_formula_callable& callable) {
+			return formula.evaluate(callable).as_int() / 1.0;
+		}));
```

Comment:
if i don't add /1.0(or other divisor) the compilation fail, and for value add and sub, thes value are individualy int, i must round these value before final calculation because a double value for me is 'illegal'

Reply:
@soliton-  I confirm that when I use static_cast<double>(def) to be able to round to the nearest integer or even convert value to double, I am forced to add a divisor to the formula, as for rounding 'value' 'add' or 'sub' individually it is because (for me) these values ​​are supposed to be integers in the final calculation, just like multiply and divide are rounded to two digits after the decimal point.

Reply:
> I confirm that when I use static_cast(def) to be able to round to the nearest integer or even convert value to double, I am forced to add a divisor to the formula

What forces you to divide instead of cast? Dividing by 1.0 is just an odd way to cast?

Reply:
replaced by round()